pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -337,7 +337,7 @@ def model_to_device(self):
 
     def pre_backward(self, closure_loss: torch.Tensor) -> None:
         """Run before precision plugin executes backward"""
-        if not self.lightning_module.automatic_optimization and self.model.require_backward_grad_sync:
```

Comment:
Why was this breaking manual opt?

Do we not need the check anymore? Why?

Reply:
Because `require_backward_grad_sync` is only part of `DistributedDataParallel`. You could see the error in #6503